src/main/kotlin/com/exactpro/th2/common/utils/message/transport/MessageBatcher.kt

```diff
@@ -30,17 +30,17 @@ import kotlin.concurrent.withLock
 class MessageBatcher(
     private val maxBatchSize: Int = 1000,
     private val maxFlushTime: Long = 1000,
-    private val book: String,
+    private val defaultBook: String,
     private val batchSelector: (Message.Builder<*>, String) -> Any = GROUP_SELECTOR,
     private val executor: ScheduledExecutorService,
     private val onError: (Throwable) -> Unit = {},
     private val onBatch: (GroupBatch) -> Unit
 ) : AutoCloseable {
     private val batches = ConcurrentHashMap<Any, Batch>()
 
-    fun onMessage(message: Message.Builder<*>, sessionGroup: String) {
+    fun onMessage(message: Message.Builder<*>, sessionGroup: String, book: String? = null) {
         message.idBuilder().setTimestamp(Instant.now())
-        batches.getOrPut(batchSelector(message, sessionGroup)) { Batch(book, sessionGroup) }
+        batches.getOrPut(batchSelector(message, sessionGroup)) { Batch(book ?: defaultBook, sessionGroup) }
```

Comment:
We can't do that because batcher batches message by session group only and expects than book is the same for all messages